### DISCUSSION TITLE
What am I doing wrong with this query?

### DISCUSSION THREAD

**Discussion by pangeasi** [trpc/trpc#2601] [Q&A]
Hello, I have been using tRPC for a short time and the truth is that I find it wonderful.
Something has happened to me that I do not understand, when making a query I am receiving a duplicate response, but it is also repeated in the url
![image](https://user-images.githubusercontent.com/28918459/187470255-ba5c11ae-fe3a-4063-a671-7627d6bbb518.png)

this only happens when strict mode is enabled, when i disable it behaves fine

![image](https://user-images.githubusercontent.com/28918459/187471017-e530fb24-8428-4244-9fed-9ce02bd4ec94.png)
what am I doing wrong?

thanks :smile: 

PD this is the repo https://github.com/pangeasi/multi-live and where query EP is https://github.com/pangeasi/multi-live/blob/main/server/user/index.ts

**Comment by KATT** [comment]
Link the file(s) where we can find `user.me` :)

**Reply by pangeasi** [reply]
yeah, its here https://github.com/pangeasi/multi-live/blob/main/server/user/index.ts in line 81, yes it requesting both i18n.t and user.me, but user.me is duplicated when strict mode is enable

**Reply by KATT** [reply]
It makes actually zero sense to me why this would happen:

- `react-query` _should_ take care of deduplication as they have the same cache key, but it doesn't seem to be working here
- If it has some re-render bug, `react-query` should cancel the previous query ahead of the next one

My guess is that it gets re-rendered several times and the query cancellation somehow doesn't work.

**Reply by KATT** [reply]
Curious to see if it's the same if you do the bare minimum [to migrate to v10](https://trpc.io/docs/v10/migrate-from-v9-to-v10)

**Reply by pangeasi** [reply]
ok, I'm trying to migrate I'll cross my fingers 🤞

**Reply by pangeasi** [reply]
Hi again @KATT I have tried this issue in v10 and finally this repetition does not happen in batching

I like to much v10! congrats and thank you for help 👍️

**Comment by k-ahlawat** [comment]
@KATT Hey, I know this is a really old discussion, but for us this is still persistent in v11 with the classic React Query client. With Strict Mode, the requests get fired twice.

If we manually do `useQuery` with an explicit `queryFn` that calls out to `trpcUtils.mypath.query`, then this issue does not happen, even with Strict Mode.

We're using the `httpBatchLink` and `abortOnUnmount` is set to true, but we don't seem to be seeing this unmount behavior.
